addons/api/tests/unit/models/host-catalog-test.js

```diff
@@ -19,4 +19,75 @@ module('Unit | Model | host catalog', function (hooks) {
     assert.true(modelA.isStatic);
     assert.false(modelB.isStatic);
   });
+
+  test('it has isPlugin function and returns the expected values', async function (assert) {
+    assert.expect(3);
+    const store = this.owner.lookup('service:store');
+    const modelPlugin = store.createRecord('host-catalog', { type: 'plugin' });
+    const modelStatic = store.createRecord('host-catalog', { type: 'static' });
+    assert.equal(typeof modelPlugin.isPlugin, 'boolean');
+    assert.true(modelPlugin.isPlugin);
+    assert.false(modelStatic.isPlugin);
+  });
+
+  test('it has isAWS function and returns the expected values', async function (assert) {
+    assert.expect(3);
+    const store = this.owner.lookup('service:store');
+    const modelAws = store.createRecord('host-catalog', {
+      plugin: { name: 'aws' },
+    });
+    const modelRandom = store.createRecord('host-catalog', {
+      plugin: { name: 'random' },
+    });
+    assert.equal(typeof modelAws.isAWS, 'boolean');
+    assert.true(modelAws.isAWS);
+    assert.false(modelRandom.isAWS);
+  });
+
+  test('it has isAzure function and return the expected values', async function (assert) {
+    assert.expect(3);
+    const store = this.owner.lookup('service:store');
+    const modelAzure = store.createRecord('host-catalog', {
+      plugin: { name: 'azure' },
+    });
+    const modelRandom = store.createRecord('host-catalog', {
+      plugin: { name: 'random' },
+    });
+    assert.equal(typeof modelAzure.isAzure, 'boolean');
+    assert.true(modelAzure.isAzure);
+    assert.false(modelRandom.isAzure);
+  });
+
+  test('get compositeType returns expected values', async function (assert) {
+    assert.expect(3);
+    const store = this.owner.lookup('service:store');
+    const modelPlugin = store.createRecord('host-catalog', {
+      type: 'plugin',
+      plugin: { name: 'Test name' },
+    });
+    const modelStatic = store.createRecord('host-catalog', {
+      type: 'static',
+    });
+    assert.equal(typeof modelPlugin.compositeType, 'string');
+    assert.equal(modelPlugin.compositeType, 'Test name');
+    assert.equal(modelStatic.compositeType, 'static');
+  });
+
+  test('set compositeType sets expected values', async function (assert) {
+    assert.expect(3);
+    const store = this.owner.lookup('service:store');
+    const modelPlugin = store.createRecord('host-catalog', {
+      type: 'plugin',
+      plugin: { name: 'Test name' },
+    });
+    const modelStatic = store.createRecord('host-catalog', {
+      type: 'static',
+    });
+    modelPlugin.set('compositeType', 'aws');
```

Comment:
@randallmorey I could not find a pattern to test set in our codebase. I found this in a old ember documentation. Thoughts?